ui/src/app/services/data/data.service.ts

```diff
@@ -0,0 +1,270 @@
+    this.fetch.getSchemaConversionFromSessionFile(payload).subscribe((res: IConv) => {
+      this.convSubject.next(res)
+      // this.initiateSession()
+    })
+  }
+  getSchemaConversionFromResumeSession(versionId: string) {
+    this.fetch.resumeSession(versionId).subscribe((res: IConv) => {
+      this.convSubject.next(res)
+    })
+  }
+
+  getRateTypemapAndSummary() {
+    return forkJoin({
+      rates: this.fetch.getConversionRate(),
+      typeMap: this.fetch.getTypeMap(),
+      summary: this.fetch.getSummary(),
+      ddl: this.fetch.getDdl(),
+    })
+      .pipe(
+        catchError((err: any) => {
+          console.log(err)
+          return of(err)
+        })
+      )
+      .subscribe(({ rates, typeMap, summary, ddl }: any) => {
+        this.conversionRateSub.next(rates)
+        this.typeMapSub.next(typeMap)
+        this.summarySub.next(new Map<string, ISummary>(Object.entries(summary)))
+        this.ddlSub.next(ddl)
+      })
+  }
+
+  updateTable(tableName: string, data: IUpdateTable): Observable<string> {
+    return this.fetch.updateTable(tableName, data).pipe(
+      catchError((e: any) => {
+        return of({ error: e.error })
+      }),
+      tap(console.log),
+      map((data: any) => {
+        if (data.error) {
+          return data.error
+        } else {
+          this.convSubject.next(data)
+          return ''
+        }
+      })
+    )
+  }
+
+  updateFkNames(tableName: string, updatedFkNames: Record<string, string>): Observable<string> {
+    return this.fetch.updateFk(tableName, updatedFkNames).pipe(
+      catchError((e: any) => {
+        return of({ error: e.error })
+      }),
+      tap(console.log),
+      map((data: any) => {
+        if (data.error) {
+          return data.error
+        } else {
+          this.convSubject.next(data)
+          return ''
+        }
+      })
+    )
+  }
+
+  dropFk(tableName: string, pos: number) {
+    return this.fetch.removeFk(tableName, pos).pipe(
+      catchError((e: any) => {
+        return of({ error: e.error })
+      }),
+      tap(console.log),
+      map((data: any) => {
+        if (data.error) {
+          return data.error
+        } else {
+          this.convSubject.next(data)
+          return ''
+        }
+      })
+    )
+  }
+
+  getConfig() {
+    this.fetch.getSpannerConfig().subscribe((res: ISpannerConfig) => {
+      this.configSub.next(res)
+    })
+  }
+
+  updateConfig(config: ISpannerConfig) {
+    this.configSub.next(config)
+  }
+
+  initiateSession() {
+    this.fetch.InitiateSession().subscribe((data: any) => {
+      console.log('get initiate session', data)
+      this.currentSessionSub.next(data)
+    })
+  }
+
+  updateGlobalType(types: Record<string, string>): void {
+    this.fetch.updateGlobalType(types).subscribe({
+      next: (data: any) => {
+        this.convSubject.next(data)
+        this.snackbar.openSnackBar('Type update successful.', 'Close', 3000)
```

Comment:
The msg is not clear , maybe use something like : Global datatype updated successfully

Reply:
Addressed in PR 287